app/javascript/components/main-menu/second-level.jsx

```diff
@@ -14,34 +15,50 @@ const SecondLevel = ({
   level,
   type,
   active,
-}) => (
-  <li className={`menu-list-group-item ${items.length > 0 ? 'tertiary-nav-item-pf' : ''} ${active ? 'active' : ''}`} id={getIdByCategory(items.length > 0, id)}>
-    <a
-      href={getHrefByType(type, href, id)}
-      onClick={(event) => {
-        if (handleUnsavedChanges(type) === false) {
-          event.preventDefault();
-        }
-        return false;
-      }}
-      target={getTargetByType(type)}
+  handleSetActiveIds,
+}) => {
+  const hasSubitems = items.length > 0;
+  return (
+    <li
+      className={clsx(
+        'menu-list-group-item',
+        {
+          'tertiary-nav-item-pf': hasSubitems,
+          'is-hover': useContext(HoverContext).secondLevelId === id,
```

Comment:
This is usage of hooks is dangerous. 1st rule of rule of react hooks is that they have to defined at the top of the function to avoid potential state inconsistencies. 
```jsx
const Foo = () => {
  const { bar }  = useSomething() // same as const bar = useSomething().bar but it is preferred syntax.
  return (
  <div> ...
  )
}
```

Granted this is not the case for `useContext` but still, we should follow the rules not to confuse anyone who might use this as a inspiration in the future.

Reply:
@fhlavac : can you move the `useContext` into a separate statement after/before line 20?

@Hyperkid123 : that is what you are asking for, right?

Reply:
moved :)